### ISSUE TITLE
Docker-based Criu portable test failure related to host machine micro-arch and Semeru Image Base OS

### ISSUE THREAD

**Issue by LongyuZhang** [eclipse-openj9/openj9#16382]
Docker-based Criu portable tests results related to Host Machine Micro-Architecture and the Semeru Docker image base OS are listed as following: 

| Host Micro-Arch \ Semeru Image Base OS  | ubuntu 20|                     ubuntu22            |
|       -----------                                                  |   ------    |                    ---------             |
|                               amd                                      |     Created Image cannot restore on skylake machine [1]     |     _Pass_   | 
|                             skylake                                  |    _Pass_  |     Cannot create checkout image [2] |
|                            broadwell                               |  Cannot create checkout image [3]   |     Cannot create checkout image [3] |

[1] https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk17_j9_sanity.external_x86-64_linux_criu/55/
```
[2022-11-26T16:24:21.470Z] Performing CRIUSupport.checkpointJVM(), current thread name: main, Mon Nov 21 04:48:53 UTC 2022, System.currentTimeMillis(): 1669006133471, System.nanoTime(): 1337250638340741
[2022-11-26T16:24:21.470Z] Unhandled exception
[2022-11-26T16:24:21.470Z] Type=Illegal instruction vmState=0x00200003
[2022-11-26T16:24:21.470Z] J9Generic_Signal_Number=00000048 Signal_Number=00000004 Error_Value=00000000 Signal_Code=00000002
[2022-11-26T16:24:21.470Z] Handler1=00007F5184774590 Handler2=00007F51846CFBF0
[2022-11-26T16:24:21.470Z] RDI=000000000000001C RSI=0000000000000000 RAX=00000000000000EE RBX=00007F5184D6FBC0
[2022-11-26T16:24:21.470Z] RCX=00007F5184F9465C RDX=0000000000000000 R8=0000000000000000 R9=0000000000000000
[2022-11-26T16:24:21.470Z] R10=0000000000000022 R11=0000000000000000 R12=00007F5184D6FCC8 R13=00007F51801163C8
[2022-11-26T16:24:21.470Z] R14=0000000000016F00 R15=00007F5184CEE540
[2022-11-26T16:24:21.470Z] RIP=00007F5184FF5C2C GS=0000 FS=0000 RSP=00007F5184D6F840
[2022-11-26T16:24:21.470Z] EFlags=0000000000010246 CS=0033 RBP=0000000000000008 ERR=0000000000000000
[2022-11-26T16:24:21.470Z] TRAPNO=0000000000000006 OLDMASK=0000000000000000 CR2=0000000000000000
[2022-11-26T16:24:21.470Z] xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm3 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm4 00ff000000000000 (f: 0.000000, d: 7.063274e-304)
[2022-11-26T16:24:21.470Z] xmm5 00007f517f338108 (f: 2134081792.000000, d: 6.916326e-310)
[2022-11-26T16:24:21.470Z] xmm6 3faaa5aa5df25984 (f: 1576163712.000000, d: 5.204518e-02)
[2022-11-26T16:24:21.470Z] xmm7 0000003000000020 (f: 32.000000, d: 1.018558e-312)
[2022-11-26T16:24:21.470Z] xmm8 2020000000202020 (f: 2105376.000000, d: 5.966673e-154)
[2022-11-26T16:24:21.470Z] xmm9 ffff000000ffffff (f: 16777215.000000, d: -nan)
[2022-11-26T16:24:21.470Z] xmm10 2020000000202020 (f: 2105376.000000, d: 5.966673e-154)
[2022-11-26T16:24:21.470Z] xmm11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2022-11-26T16:24:21.470Z] xmm15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
```
[2] https://hyc-runtimes-jenkins.swg-devops.com/job/CRIU_Image_Upload/142
```
11:55:32   [OUT] Pre-checkpoint
11:55:32   [OUT] Performing CRIUSupport.checkpointJVM(), current thread name: main, Tue Nov 29 16:55:25 UTC 2022, System.currentTimeMillis(): 1669740925556, System.nanoTime(): 3283579029994489
11:55:32   [OUT] Exception in thread "main" org.eclipse.openj9.criu.SystemCheckpointException: Could not dump the JVM processes, err=-52
11:55:32   [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
11:55:32   [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVM(CRIUSupport.java:595)
11:55:32   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:76)
11:55:32   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:64)
11:55:32   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:60)
11:55:32   [OUT] 	at org.openj9.criu.CRIUSimpleTest.checkpoints(CRIUSimpleTest.java:43)
11:55:32   [OUT] 	at org.openj9.criu.CRIUSimpleTest.main(CRIUSimpleTest.java:33)
11:55:32   [OUT] finished script
```
[3] https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder/29852
```
12:04:48   [OUT] Can't exec criu swrk: Operation not permitted
12:04:48   [OUT] Can't read request: Connection reset by peer
12:04:48   [OUT] Can't receive response: Invalid argument
12:04:48   [OUT] Exception in thread "main" org.eclipse.openj9.criu.SystemCheckpointException: Could not dump the JVM processes, err=-70
12:04:48   [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
12:04:48   [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVM(CRIUSupport.java:595)
12:04:48   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:76)
12:04:48   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:64)
12:04:48   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:60)
12:04:48   [OUT] 	at org.openj9.criu.CRIUSimpleTest.checkpoints(CRIUSimpleTest.java:43)
12:04:48   [OUT] 	at org.openj9.criu.CRIUSimpleTest.main(CRIUSimpleTest.java:33)
12:04:48   [OUT] finished script
```

**Comment by LongyuZhang** [comment]
FYI @tajila @llxia

**Comment by JasonFengJ9** [comment]
> [2022-11-26T16:24:21.470Z] Type=Illegal instruction vmState=0x00200003

This appears a dup of https://github.com/eclipse-openj9/openj9/issues/16372

> Exception in thread "main" org.eclipse.openj9.criu.SystemCheckpointException: Could not dump the JVM processes, err=-70

Infra issue, to be addressed by https://github.com/eclipse-openj9/openj9/pull/16373

> Exception in thread "main" org.eclipse.openj9.criu.SystemCheckpointException: Could not dump the JVM processes, err=-52

A known issue https://github.com/eclipse-openj9/openj9/issues/15631

**Comment by tajila** [comment]
@LongyuZhang Do you have steps to manually reproduce the issue?

**Comment by tajila** [comment]
Also, what version of criu are you using in the image?

**Comment by LongyuZhang** [comment]
The criu was built from source code with branch [`criu-dev`](https://github.com/adoptium/aqa-tests/blob/983efa887d7f9d43f13cff4a06ade76554824f22/external/dockerfile_functions.sh#L394).

The manual reproduce of the restore issue are:
```
- reserve machine with corresponding micro-architecture (e.g. skylake machine)
- docker login sys-rt-docker-local.artifactory.swg-devops.com  (with your w3 id)
- docker pull sys-rt-docker-local.artifactory.swg-devops.com/criu_image_upload/17-openj9-ubuntu-linux_x86-64-hw.arch.x86.amd:143
- docker run --privileged  -v /var/run/docker.sock:/var/run/docker.sock  -v /home/jenkins/workspace/Test_openjdk17_j9_sanity.external_x86-64_linux_criu/openjdkbinary/j2sdk-image:/opt/java/openjdk  --tmpfs /run  --name restore-test --rm sys-rt-docker-local.artifactory.swg-devops.com/criu_image_upload/17-openj9-ubuntu-linux_x86-64-hw.arch.x86.amd:143
- docker logout
```

**Comment by vijaysun-omr** [comment]
I thought I'd mention @ymanton here so that this is on his radar too, even though I believe Tobi was planning to look into this more first (as discussed on the status call this morning).

**Comment by tajila** [comment]
@LongyuZhang Are your instructions for the restore run? I thought you were seeing issues on the checkpoint run

**Comment by tajila** [comment]
When I run it manually with ubuntu22 using the latest changes (privileged mode) I see the following error:

```
(00.087246) Error (criu/cr-dump.c:203): 68 has rseq but kernel lacks get_rseq_conf feature
(00.087254) ----------------------------------------
(00.087256) Error (criu/cr-dump.c:1714): Dump core (pid: 68) failed with -1
...
(00.105297) Unlock network
(00.105313) Unfreezing tasks into 1
(00.105320) 	Unseizing 68 into 1
(00.105599) Error (criu/cr-dump.c:2099): Dumping FAILED.
```

**Comment by tajila** [comment]
@LongyuZhang can you paste the contents of the log file when  it fails for you? I want to see if it matches with the failure I am seeing

**Comment by LongyuZhang** [comment]
@tajila I have updated the branch to latest and jdk to Dec.1, re-run the checkpoint create on skylake machine (ubu20) with semeru base image ubu22, the output is still the same as follows ([link here](https://hyc-runtimes-jenkins.swg-devops.com/job/CRIU_Image_Upload/146/console)), I did not see the error you pasted before.
```
19:29:01   [OUT] Performing CRIUSupport.checkpointJVM(), current thread name: main, Tue Dec 06 00:29:00 UTC 2022, System.currentTimeMillis(): 1670286540528, System.nanoTime(): 192859447690753
19:29:01   [OUT] Exception in thread "main" org.eclipse.openj9.criu.SystemCheckpointException: Could not dump the JVM processes, err=-52
19:29:01   [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
19:29:01   [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVM(CRIUSupport.java:594)
19:29:01   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:76)
19:29:01   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:64)
19:29:01   [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:60)
19:29:01   [OUT] 	at org.openj9.criu.CRIUSimpleTest.checkpoints(CRIUSimpleTest.java:43)
19:29:01   [OUT] 	at org.openj9.criu.CRIUSimpleTest.main(CRIUSimpleTest.java:33)
19:29:01   [OUT] finished script
```

**Comment by tajila** [comment]
@LongyuZhang you need to paste the contents of [checkpointDir]/logs

**Comment by LongyuZhang** [comment]
@tajila I have updated the log output on above comment. The image with failed log are also updated to hyc_jenkins docker registry if you need.
```
docker pull sys-rt-docker-local.artifactory.swg-devops.com/criu_debug_images/host_ubu20_image_os_ubu22_17:latest
docker run -it --entrypoint=sh sys-rt-docker-local.artifactory.swg-devops.com/criu_debug_images/host_ubu20_image_os_ubu22_17:latest
cd /aqa-tests/TKG/output_1670606264531/cmdLineTester_criu_keepCheckpoint_0/cpData
```

**Comment by tajila** [comment]
Thanks @LongyuZhang . We are seeing the same issue.

Can you try adding `glibc.pthread.rseq=0` glibc_tunables command?

**Comment by LongyuZhang** [comment]
The creating checkpoint works after adding  `glibc.pthread.rseq=0`. Thanks @tajila 
I will test restoring it on amd and skylake machines then.

**Comment by LongyuZhang** [comment]
Restore passed on both amd and skylake, though it shows warning and error, the restore was successful ([link here](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/30170/console)).
```
15:34:04  Warn  (criu/net.c:3427): Unable to get tun network namespace
15:34:04  Warn  (criu/sk-unix.c:224): unix: Unable to open a socket file: Bad address
15:34:04  Warn  (criu/net.c:3427): Unable to get socket network namespace
15:34:04  Warn  (criu/kerndat.c:1421): Can't get pidfd
15:34:04  pie: 888: Warn  (criu/pie/restorer.c:1380): Can't restore vvar - continuing regardless
15:34:04  Error (amdgpu_plugin.c:1931): amdgpu_plugin: failed to open kfd in plugin: No such file or directory
15:34:04  Error (amdgpu_plugin.c:1931): amdgpu_plugin: failed to open kfd in plugin: No such file or directory
15:34:04  Test cmdLineTester_criu_keepCheckpoint_0 passed
```

**Comment by tajila** [comment]
Thanks @LongyuZhang. I think @amdgpu_plugin was working on something for the amdgpu_plugin issues.

@ymanton is the rseq limitation something we need to live with?

**Comment by ymanton** [comment]
The `amdgpu_plugin` errors can be ignored, and the `rseq` issue has been fixed by https://github.com/eclipse-openj9/openj9/commit/172bcf7244116186d4edde019473db69fb5d9559 so I think we can close this?

**Comment by LongyuZhang** [comment]
Yes, we can close this issue then.

> The `amdgpu_plugin` errors can be ignored, and the `rseq` issue has been fixed by [172bcf7](https://github.com/eclipse-openj9/openj9/commit/172bcf7244116186d4edde019473db69fb5d9559) so I think we can close this?
